src/main/java/org/wso2/carbon/module/swiftiso20022/JsonToMT103Validator.java

```diff
@@ -0,0 +1,137 @@
+ */
+public class JsonToMT103Validator extends AbstractConnector {
+
+    private static final Gson gson = new Gson();
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+
+        this.log.debug("Executing JsonToMT103Transformer to convert the JSON payload to MT103 format");
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+
+        // get JSON payload as a string from the message context
+        Optional<String> payload = ConnectorUtils.buildMessagePayloadFromMessageContext(axis2MessageContext);
+
+        // if JSON payload is available
+        if (payload.isPresent()) {
+
+            // check JSON object with request model (data types of fields)
+            try {
+                validatePayloadWithMT103JsonModel(payload.get());
+            } catch (JsonSyntaxException e) {
+                this.log.error(ConnectorConstants.ERROR_INVALID_PAYLOAD, e);
+                ConnectorUtils.appendErrorToMessageContext(
+                        messageContext, ConnectorConstants.INVALID_REQUEST_PAYLOAD, e.getMessage());
+                super.handleException(ConnectorConstants.ERROR_INVALID_PAYLOAD, messageContext);
+            }
+
+            // validate values of the JSON object
+            this.log.debug("Validating payload parameters");
+            ValidationResult validationResponse = validateRequestPayload(payload.get());
+            if (validationResponse.isNotValid()) {
+                this.log.error(validationResponse.getErrorMessage());
+                ConnectorUtils.appendErrorToMessageContext(messageContext, validationResponse.getErrorCode(),
+                        validationResponse.getErrorMessage());
+                super.handleException(ConnectorConstants.ERROR_VALIDATION_FAILED, messageContext);
+            }
+        } else {
+            this.log.error(ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            ConnectorUtils.appendErrorToMessageContext(messageContext, ConnectorConstants.MISSING_REQUEST_PAYLOAD,
+                    ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            super.handleException(ConnectorConstants.ERROR_MISSING_PAYLOAD, messageContext);
+        }
+    }
+
+    /**
+     * Method to validate each block in the request payload.
+     *
+     * @param payloadString string containing JSON payload.
+     * @return empty or validation result with an error message.
+     */
+    private ValidationResult validateRequestPayload(String payloadString) {
+
+        // build JSONObject from the payload string
+        JSONObject requestPayload = new JSONObject(payloadString);
+
+        // validate block 01
+        ValidationResult blockValidationResult =
+                JsonToMt103Utils.validateBlock01(requestPayload.optJSONObject(MT103Constants.BLOCK01));
+        if (blockValidationResult.isNotValid()) {
+            return blockValidationResult;
```

Comment:
Shall we add debug logs before returning the result?

Reply:
Debug logs added in commit 0ed9e610c347551ceb811aad8f3b92789ceda026